composer/trainer/trainer.py

```diff
@@ -1739,16 +1739,14 @@ def _eval_train_metrics(self, device_batch):
                         found_cuda_oom = 1
                     else:
                         raise
-                # Auto grad accum only supported on GPU
-                if isinstance(self._device, DeviceGPU):
-                    # Propagate across all ranks if any rank hit CUDA OOM
-                    found_cuda_oom = self._device.tensor_to_device(torch.tensor([found_cuda_oom], dtype=torch.uint8))
-                    dist.all_reduce(found_cuda_oom, reduce_operation='MAX')
-                    if found_cuda_oom.item() == 1:
-                        device_batch_size = self._train_data_spec.get_num_samples_in_batch(device_batch)
-                        _adjust_eval_batch_split(self.state, device_batch_size)
-                        # Skip return and rerun after handling oom
-                        continue
+                # Propagate across all ranks if any rank hit CUDA OOM
+                found_cuda_oom = self._device.tensor_to_device(torch.tensor([found_cuda_oom], dtype=torch.uint8))
```

Comment:
I find this still a bit odd, that we are doing `found_cuda_oom` reductions for _any_ hardware device (even non-CUDA ones). Since `found_cuda_oom` will always be zero for non-GPU devices, why not add a guard here to skip this step?

Reply:
Updated. PR now ensures check is same as guard (or it will break for same reason described in description).

Reply:
NVM I see the TPU check, but I think this is still cleaner + now properly handles seq length warmup